File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03989/p03989_num4_parsed.cpp
#include <iostream>
#include <vector>
using namespace std;

const int MOD = 924844033;

int main() {
    int N, K;
    cin >> N >> K;

    vector<vector<int>> dp(N + 1, vector<int>(N + 1, 0));
    dp[0][0] = 1;

    for (int i = 1; i <= N; i++) {
        for (int j = 0; j <= N; j++) {
            dp[i][j] = (dp[i][j] + dp[i - 1][j] + (j > 0 ? dp[i - 1][j - 1] : 0)) % MOD;
            if (j + K + 1 <= N) {
                dp[i][j + K + 1] = (dp[i][j + K + 1] - dp[i - 1][j] + MOD) % MOD;
            }
        }
    }

    int answer = 0;
    for (int i = 0; i <= N; i++) {
        answer = (answer + dp[N][i]) % MOD;
    }

    cout << answer << endl;

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03989/p03989_num4_parsed.py
MOD = 924844033

def count_permutations(N, K):
    dp = [0] * (N + 1)
    dp[0] = 1
     
    for i in range(N):
        dp[i + 1] = (dp[i + 1] - dp[i] + dp[i]) % MOD
        for j in range(2 * K + 1):
            if i + j + 2 <= N:
                dp[i + j + 2] = (dp[i + j + 2] + dp[i]) % MOD
     
    return dp[N]

N, K = map(int, input().split())
answer = count_permutations(N, K)
print(answer)